### PR TITLE
Finalize user_agent migration to ECS

### PR DESCRIPTION
**Pull request by webmat**
This PR is built on top of #10472 and should be merged after it.

In 7.0, the output of ingest node "user_agent" processor will match ECS by default (https://github.com/elastic/elasticsearch/pull/37984, https://github.com/elastic/elasticsearch/issues/38094). This means we can remove a lot of temporary field renames that were previously needed, to make its output follow ECS.

This PR will fail CI until the ES Docker image used in CI includes the changes from both issues linked above. Both of them are merged to master already, so we're just waiting for the next successful artifact build for ES.

Filebeat datasets affected:

- apache.access
- iis.access
- nginx.access
- suricata.eve (x-pack)
- traefik.access
- zeek.http (x-pack)

This PR also fixes a few mistakes that were made in ecs-migration.yml.

**Caveat**

This PR introduces multiple changes / removals of the `@timestamp` field in the test files because of an unrelated change in how Elasticsearch parses timestamps. This will be fixed in other PRs.

**TODO**

- [x] Get rid of both sets of major/minor/patch fields (ua.* and ua.os.*) in fields.yml
  - delete the fields
  - delete the aliases to these fields
  - ecs-migration.yml entries now refer to the corresponding .version field, with `alias: false`
- [x] Correct a few mistakes in ecs-migration.yml
  - Add missing entries
  - Correctly point the device field to ua.device.name
  - Remove incorrect entries (user_name under user_agent)
  - Remove obsolete comment
- [x] Reformat each set of entries to be in the same order
- [x] Rebase over #10472 once merged
- [ ] Confirm build is green, once CI runs on Elasticsearch that includes this PR https://github.com/elastic/elasticsearch/pull/38115

### CODE REVIEW

File: filebeat/module/apache/access/ingest/default.json [hunk 1]
```diff
@@ -39,45 +39,9 @@
     }
   }, {
     "user_agent": {
-      "field": "apache.access.agent",
+      "field": "user_agent.original",
       "ignore_failure": true
     }
-  }, {
```
Comment (ruflin) [inline filebeat/module/apache/access/ingest/default.json]: Nice to see that some much code can be removed.

**Comment by webmat** [comment]
I'm trying to make this PR work for the user_agent without this fix https://github.com/elastic/elasticsearch/issues/38094. Not sure if CI incorporates yesterday's change, which causes a mapping error. If it does, I want this PR to be merged to resolve the issue, and I'll do a follow-up PR once the final fix is in, to remove the last rename.

**Comment by webmat** [comment]
The fun meter raises by another level: I just noticed that the ECS definition of `user_agent.device.name` was imported, but then was later overridden by a "temporary" field definition, defining `user_agent.device` as `keyword`...

File: filebeat/module/iis/access/test/test-iis-7.5.log-expected.json [hunk 2]
```diff
@@ -22,11 +22,9 @@
         "url.query": "-",
         "user.name": "-",
         "user_agent.device.name": "Other",
-        "user_agent.major": "7",
-        "user_agent.minor": "0",
         "user_agent.name": "IE",
         "user_agent.original": "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.3; WOW64; Trident/7.0; .NET4.0E; .NET4.0C; .NET CLR 3.5.30729; .NET CLR[ 2.0.50727](tel: 2050727); .NET CLR 3.0.30729)",
-        "user_agent.os.full_name": "Windows 8.1",
-        "user_agent.os.name": "Windows 8.1"
+        "user_agent.os.name": "Windows 8.1",
```
Comment (ruflin) [inline filebeat/module/iis/access/test/test-iis-7.5.log-expected.json]: Is this expected? I see below this goes under `full`?

**Comment by ruflin** [comment]
Canceled Windows jobs can be ignored.

**Comment by ycombinator** [comment]
> For the timestamp changes I would suggest we ignore it for now to make sure CI is green as we already work on fixing it. @ycombinator objections?

No objections from me. I'm +1 to deal with those changes in separate PR(s).

**Comment by ruflin** [comment]
@webmat @ycombinator Let's make sure that any follow up PR to fix the timestamp changes / fixes the timestamps we have remove / changed here.

**Comment by webmat** [comment]
It's mostly on the ES side, though.

We may proactively be able to circumvent some of parsing problems in our pipelines. But since they're regressions in ts parsing, they should also be fixed in ES.

**Comment by ycombinator** [comment]
> We may proactively be able to circumvent some of parsing problems in our pipelines. But since they're regressions in ts parsing, they should also be fixed in ES.

++ to trying to fix on the ES side. I've been trying to adapt our pipelines in a way that would work with `.` or `,` as the delimiter between seconds and milliseconds as well as with timestamps that do or do not contain time zone offsets in them. I haven't found a way to make all 4 of those cases work as expected yet. I'll put up a PR to demonstrate the challenges.

**Comment by webmat** [comment]
@ycombinator Yes, even if never merged, would make it easy for me to 👀 .

Otherwise I doubt I'll get to that in time :-\

**Comment by ycombinator** [comment]
@webmat https://github.com/elastic/beats/pull/10543

**Pull request merged**